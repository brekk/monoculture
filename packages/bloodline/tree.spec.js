import {
  getId,
  groupTree,
  plant,
  flattenTree,
  rootedPlant,
  familyTree,
} from './tree'
// import { sep as SEP } from 'node:path'
// import { nthIndex } from 'knot'

const tree = plant({}, '../..', '../monocle/cli.js')
const sharedConfig = { basePath: '../../../', includeNpm: true }
const rooted = rootedPlant(
  { ...sharedConfig, includeNpm: false },
  '../..',
  '../monocle/cli.js'
)
const rootedWithNpm = rootedPlant(sharedConfig, '../..', '../monocle/cli.js')

test('getId', () => {
  expect(getId('..', {}, 'a')).toEqual('bloodline/a')
})

test('groupTree', () => {
  const pFlatTree = groupTree(sharedConfig, {}, {}, tree)
  expect(pFlatTree).toMatchSnapshot()
})

test('groupTree - rooted', () => {
  const pFlatTree = groupTree(sharedConfig, {}, {}, rooted)
  expect(pFlatTree).toMatchSnapshot()
})
test('groupTree - rooted with npm', () => {
  const pFlatTree = groupTree(sharedConfig, {}, {}, rootedWithNpm)
  expect(pFlatTree).toMatchSnapshot()
})

test('flattenTree', () => {
  const flatTree = flattenTree(sharedConfig, {}, {}, tree)
  expect(flatTree).toMatchSnapshot()
})

test('familyTree', () => {
  const family = familyTree(sharedConfig, {}, {}, tree)
  expect(family).toMatchSnapshot()
})
