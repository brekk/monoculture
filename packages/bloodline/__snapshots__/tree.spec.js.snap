// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flattenTree 1`] = `
{
  "monoculture/packages/monocle/cli.js": [
    "monoculture/node_modules/chalk/source/index.js",
    "monoculture/node_modules/chalk/source/utilities.js",
    "monoculture/node_modules/chalk/source/vendor/ansi-styles/index.js",
    "monoculture/node_modules/ramda/es/identity.js",
    "monoculture/node_modules/ramda/es/internal/_curry1.js",
    "monoculture/node_modules/ramda/es/internal/_identity.js",
    "monoculture/node_modules/ramda/es/always.js",
    "monoculture/node_modules/ramda/es/internal/_curry1.js",
    "monoculture/node_modules/ramda/es/uniqBy.js",
    "monoculture/node_modules/ramda/es/internal/_Set.js",
    "monoculture/node_modules/ramda/es/internal/_curry2.js",
    "monoculture/node_modules/ramda/es/internal/_dispatchable.js",
    "monoculture/node_modules/ramda/es/internal/_xuniqBy.js",
    "monoculture/node_modules/ramda/es/curry.js",
    "monoculture/node_modules/ramda/es/internal/_curry1.js",
    "monoculture/node_modules/ramda/es/curryN.js",
    "monoculture/node_modules/ramda/es/mergeRight.js",
    "monoculture/node_modules/ramda/es/internal/_objectAssign.js",
    "monoculture/node_modules/ramda/es/internal/_curry2.js",
    "monoculture/node_modules/ramda/es/pipe.js",
    "monoculture/node_modules/ramda/es/internal/_arity.js",
    "monoculture/node_modules/ramda/es/internal/_pipe.js",
    "monoculture/node_modules/ramda/es/reduce.js",
    "monoculture/node_modules/ramda/es/tail.js",
    "monoculture/node_modules/ramda/es/chain.js",
    "monoculture/node_modules/ramda/es/internal/_curry2.js",
    "monoculture/node_modules/ramda/es/internal/_dispatchable.js",
    "monoculture/node_modules/ramda/es/internal/_makeFlat.js",
    "monoculture/node_modules/ramda/es/internal/_xchain.js",
    "monoculture/node_modules/ramda/es/map.js",
    "monoculture/node_modules/ramda/es/map.js",
    "monoculture/node_modules/ramda/es/internal/_arrayReduce.js",
    "monoculture/node_modules/ramda/es/internal/_curry2.js",
    "monoculture/node_modules/ramda/es/internal/_dispatchable.js",
    "monoculture/node_modules/ramda/es/internal/_map.js",
    "monoculture/node_modules/ramda/es/internal/_xmap.js",
    "monoculture/node_modules/ramda/es/curryN.js",
    "monoculture/node_modules/ramda/es/keys.js",
    "monoculture/node_modules/ramda/es/length.js",
    "monoculture/node_modules/ramda/es/internal/_curry1.js",
    "monoculture/node_modules/ramda/es/internal/_isNumber.js",
    "monoculture/node_modules/fluture/index.cjs",
    "monoculture/node_modules/file-system/index.js",
    "monoculture/node_modules/file-system/find-up.js",
    "monoculture/node_modules/file-system/exec.js",
    "monoculture/node_modules/file-system/fs.js",
    "monoculture/node_modules/file-system/interpret.js",
    "monoculture/node_modules/file-system/path.js",
    "monoculture/packages/monocle/reader.js",
    "monoculture/node_modules/ramda/src/index.js",
    "monoculture/node_modules/fluture/index.cjs",
    "monoculture/node_modules/file-system/index.js",
    "monoculture/packages/monocle/hash.js",
    "monoculture/packages/monocle/trace.js",
    "monoculture/packages/monocle/package.json",
    "monoculture/packages/monocle/trace.js",
    "monoculture/node_modules/envtrace/envtrace.js",
    "monoculture/packages/monocle/config.js",
    "monoculture/node_modules/climate-toml/climate-toml.js",
    "monoculture/node_modules/ramda/src/index.js",
    "monoculture/node_modules/climate-json/climate-json.js",
    "monoculture/node_modules/ramda/src/index.js",
  ],
}
`;

exports[`groupTree - rooted 1`] = `
{
  "monoculture/packages/bloodline/npmPaths": [],
  "monoculture/packages/bloodline/tree": [
    [
      "monoculture/packages/monocle/cli.js",
    ],
  ],
}
`;

exports[`groupTree - rooted with npm 1`] = `
{
  "monoculture/packages/bloodline/npmPaths": [
    [
      "monoculture/packages/monocle/cli.js",
      "monoculture/packages/bloodline/0",
      "monoculture/packages/bloodline/1",
      "monoculture/packages/bloodline/2",
      "monoculture/packages/bloodline/3",
      "monoculture/packages/bloodline/4",
      "monoculture/packages/bloodline/5",
      "monoculture/packages/bloodline/6",
      "monoculture/packages/bloodline/7",
      "monoculture/packages/bloodline/8",
      "monoculture/packages/bloodline/9",
      "monoculture/packages/bloodline/10",
      "monoculture/packages/bloodline/11",
      "monoculture/packages/bloodline/12",
      "monoculture/packages/bloodline/13",
    ],
  ],
  "monoculture/packages/bloodline/tree": [
    [
      "monoculture/packages/monocle/cli.js",
    ],
  ],
}
`;

exports[`groupTree 1`] = `
{
  "monoculture/packages/monocle/cli.js": [
    [
      "monoculture/node_modules/chalk/source/index.js",
      "monoculture/node_modules/chalk/source/utilities.js",
      "monoculture/node_modules/chalk/source/vendor/ansi-styles/index.js",
    ],
    [
      "monoculture/node_modules/ramda/es/identity.js",
      "monoculture/node_modules/ramda/es/internal/_curry1.js",
      "monoculture/node_modules/ramda/es/internal/_identity.js",
    ],
    [
      "monoculture/node_modules/ramda/es/always.js",
      "monoculture/node_modules/ramda/es/internal/_curry1.js",
    ],
    [
      "monoculture/node_modules/ramda/es/uniqBy.js",
      "monoculture/node_modules/ramda/es/internal/_Set.js",
      "monoculture/node_modules/ramda/es/internal/_curry2.js",
      "monoculture/node_modules/ramda/es/internal/_dispatchable.js",
      "monoculture/node_modules/ramda/es/internal/_xuniqBy.js",
    ],
    [
      "monoculture/node_modules/ramda/es/curry.js",
      "monoculture/node_modules/ramda/es/internal/_curry1.js",
      "monoculture/node_modules/ramda/es/curryN.js",
    ],
    [
      "monoculture/node_modules/ramda/es/mergeRight.js",
      "monoculture/node_modules/ramda/es/internal/_objectAssign.js",
      "monoculture/node_modules/ramda/es/internal/_curry2.js",
    ],
    [
      "monoculture/node_modules/ramda/es/pipe.js",
      "monoculture/node_modules/ramda/es/internal/_arity.js",
      "monoculture/node_modules/ramda/es/internal/_pipe.js",
      "monoculture/node_modules/ramda/es/reduce.js",
      "monoculture/node_modules/ramda/es/tail.js",
    ],
    [
      "monoculture/node_modules/ramda/es/chain.js",
      "monoculture/node_modules/ramda/es/internal/_curry2.js",
      "monoculture/node_modules/ramda/es/internal/_dispatchable.js",
      "monoculture/node_modules/ramda/es/internal/_makeFlat.js",
      "monoculture/node_modules/ramda/es/internal/_xchain.js",
      "monoculture/node_modules/ramda/es/map.js",
    ],
    [
      "monoculture/node_modules/ramda/es/map.js",
      "monoculture/node_modules/ramda/es/internal/_arrayReduce.js",
      "monoculture/node_modules/ramda/es/internal/_curry2.js",
      "monoculture/node_modules/ramda/es/internal/_dispatchable.js",
      "monoculture/node_modules/ramda/es/internal/_map.js",
      "monoculture/node_modules/ramda/es/internal/_xmap.js",
      "monoculture/node_modules/ramda/es/curryN.js",
      "monoculture/node_modules/ramda/es/keys.js",
    ],
    [
      "monoculture/node_modules/ramda/es/length.js",
      "monoculture/node_modules/ramda/es/internal/_curry1.js",
      "monoculture/node_modules/ramda/es/internal/_isNumber.js",
    ],
    [
      "monoculture/node_modules/fluture/index.cjs",
    ],
    [
      "monoculture/node_modules/file-system/index.js",
      "monoculture/node_modules/file-system/find-up.js",
      "monoculture/node_modules/file-system/exec.js",
      "monoculture/node_modules/file-system/fs.js",
      "monoculture/node_modules/file-system/interpret.js",
      "monoculture/node_modules/file-system/path.js",
    ],
    [
      "monoculture/packages/monocle/reader.js",
      "monoculture/node_modules/ramda/src/index.js",
      "monoculture/node_modules/fluture/index.cjs",
      "monoculture/node_modules/file-system/index.js",
      "monoculture/packages/monocle/hash.js",
      "monoculture/packages/monocle/trace.js",
    ],
    [
      "monoculture/packages/monocle/package.json",
    ],
    [
      "monoculture/packages/monocle/trace.js",
      "monoculture/node_modules/envtrace/envtrace.js",
    ],
    [
      "monoculture/packages/monocle/config.js",
    ],
    [
      "monoculture/node_modules/climate-toml/climate-toml.js",
      "monoculture/node_modules/ramda/src/index.js",
    ],
    [
      "monoculture/node_modules/climate-json/climate-json.js",
      "monoculture/node_modules/ramda/src/index.js",
    ],
  ],
}
`;
